### PR TITLE
Add/Change PHP.MigrationSet to update array/list syntax to short one

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

**Comment by tigitz** [comment]
I understand "Migration" as a replacement of a deprecated concept by a new one.

Short array syntax is just an alternative way to express your arrays. `array()` is not deprecated and short array syntax is not meant to replace it.

Just because someone prefers `array()` syntax doesn't mean they haven't migrated to 5.4

I feel it the rule should stay out of a migration set

**Comment by keradus** [comment]
When I created Migration rulesets, they were for purpose of "let use the new language/syntax/feature of higher version of PHP/PHPUnit". I agree one can say it's a matter of style. But one can also say that not using `**` operator is matter of style, and we aim to use it, if possible. But no worry, one can always disable the rules s/he would not like to follow

**Comment by keradus** [comment]
Closes #5389

File: doc/ruleSets/PHP54Migration.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+============================
+Rule set ``@PHP54Migration``
+============================
+
+Rules to improve code for PHP 5.6 compatibility.
```
Comment (glensc) [inline doc/ruleSets/PHP54Migration.rst:5]: PHP54Migration but then you say PHP 5.6, seems weird.

Comment (keradus) [reply doc/ruleSets/PHP54Migration.rst:5]: typo, already fixed in other PR. thanks

**Pull request merged**